scripts/build-package.js

```diff
@@ -101,57 +96,27 @@ function run() {
   } else {
     // hits here when running yarn build --packagename
     watchMode = process.argv.includes('--watch');
-    selection = Promise.resolve(
-      Object.keys(tasks)
-        .map((key) => tasks[key])
-        .filter((item) => item.name !== 'watch' && item.value === true)
-    );
+    selection = Object.keys(tasks)
+      .map((key) => tasks[key])
+      .filter((item) => item.name !== 'watch' && item.value === true);
   }
 
-  selection
-    .then((list) => {
-      const packageNames = list
-        // filters out watch command if --watch is used
-        .map((key) => key.suffix)
-        .filter(Boolean);
-
-      let glob =
-        packageNames.length > 1
-          ? `@storybook/{${packageNames.join(',')}}`
-          : `@storybook/${packageNames[0]}`;
-
-      const isAllPackages = process.argv.includes('--all');
-      if (isAllPackages) {
-        glob = '@storybook/*';
-
-        log.warn(
-          'You are building a lot of packages on watch mode. This is an expensive action and might slow your computer down.\nIf this is an issue, run yarn build to filter packages and speed things up!'
-        );
-      }
-
-      if (watchMode) {
-        const runWatchMode = () => {
-          const baseWatchCommand = `lerna exec --scope '${glob}' --parallel -- cross-env-shell node ${resolve(
-            __dirname
-          )}`;
-          const watchTsc = `${baseWatchCommand}/utils/watch-tsc.js`;
-          const watchBabel = `${baseWatchCommand}/utils/watch-babel.js`;
-          const command = `concurrently --kill-others-on-fail "${watchTsc}" "${watchBabel}"`;
-
-          spawn(command);
-        };
+  selection.filter(Boolean).forEach((v) => {
+    const sub = execa.command(`yarn prepare${watchMode ? ' --watch' : ''}`, {
+      cwd: resolve(__dirname, '..', v.location),
+      buffer: false,
+    });
 
-        runWatchMode();
-      } else {
-        spawn(`lerna run prepare --scope "${glob}"`);
-      }
-      process.stdout.write('\x07');
-    })
-    .catch((e) => {
-      log.aborted(prefix, chalk.red(e.message));
-      log.silly(prefix, e);
-      process.exit(1);
+    sub.stdout.on('data', (data) => {
+      process.stdout.write(`${chalk.cyan(v.name)}:\n${data}`);
```

Comment:
prefix each chunk of output, with the package where it came from, make the name red, if it came from `stdout`.